### PR TITLE
Add a workflow to run experiments for Beam

### PR DESCRIPTION
**Pull request by clayburn**

### CODE REVIEW

File: .github/workflows/run-experiments-apache-beam.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        if: matrix.experimentId == 1
+      - name: Run experiment 2
+        uses: gradle/gradle-enterprise-build-validation-scripts/.github/actions/gradle/experiment-2@actions-stable
+        env:
+          GRADLE_ENTERPRISE_ACCESS_KEY: "${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}"
+        with:
+          gitRepo: ${{ env.GIT_REPO }}
+          tasks: ${{ env.TASKS }}
+          gradleEnterpriseUrl: ${{ env.GRADLE_ENTERPRISE_URL }}
+          failIfNotFullyCacheable: true # Require no issues to be found
```
Comment (erichaagdev) [inline .github/workflows/run-experiments-apache-beam.yml:66]: Do we need the comment?

File: .github/workflows/run-experiments-apache-beam.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: 8
+          distribution: "temurin"
+      - name: Download latest version of the validation scripts
+        uses: gradle/gradle-enterprise-build-validation-scripts/.github/actions/gradle/download@actions-stable
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Remove Maven Settings
+        run: rm $HOME/.m2/settings.xml
```
Comment (erichaagdev) [inline .github/workflows/run-experiments-apache-beam.yml:42]: What is this here for? Isn't this a Gradle build?

**Review by erichaagdev — CHANGES_REQUESTED** [review]
Left a few comments.

File: .github/workflows/run-experiments-apache-beam.yml (hunk 1 above)
Comment (clayburn) [reply .github/workflows/run-experiments-apache-beam.yml:66]: No, I'm not sure where I got this from. I'll remove it (and may use this PR to remove it from other ASF experiments)

**Review by erichaagdev — APPROVED** [review]
👍 👍

**Pull request merged**